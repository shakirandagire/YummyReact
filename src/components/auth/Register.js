import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import axios from 'axios';
import { notify } from 'react-notify-toast';

import { BASE_URL } from '../../../src/constants';
/**
 * Component for registering the users of the application
 */
class Register extends Component {
    state = {
      email: '',
      password: '',
      security_question: '',
      security_answer: '',
    }
  // Method for handling the input change
    handleInputChange = (event) => {
      const { name, value } = event.target;
      this.setState({ [name]: value });
    }
    // Method for handling the registartion of users
    handleRegister = (event) => {
      const { email, password, security_question, security_answer } = this.state;
      const data = {
        email,
        password,
        security_question,
        security_answer,

      };
      event.preventDefault();
      axios
        .post(`${BASE_URL}/api/v1/auth/register`, data)
        .then((response) => {
          notify.show(response.data.message, 'success', 4000);
          this.props
            .history
            .push('/login');
        })
        .catch((error) => {
          if (error.response) {
            notify.show(error.response.data.message, 'success');
          } else if (error.request) {
            notify.show("Request not made");
          }
        });
    }

    render() {
      const { email, password, security_question, security_answer } = this.state;

      return (
        <div className="my_container backgimg" >
          <nav className="navbar navbar-inverse">
            <div className="container-fluid">
              <div className="navbar-header">
                <Link className="navbar-brand title" to="/">Yummy Recipe App</Link>
              </div>
            </div>
          </nav>

          <div className="mycontent register">

            <p> Donot have an account with us:</p>
            <form onSubmit={this.handleRegister}>

              <div className="form-group">
                Email:
                <input type="email" name="email" className="form-control email" required onChange={this.handleInputChange} value={email} />
              </div>
              <div className="form-group">
                Password:
                <input type="password" name="password" className="form-control password" required onChange={this.handleInputChange} value={password} />
              </div>
              <div className="form-group">
                Security Question:
                <input type="text" name="security_question" className="form-control security_question" required onChange={this.handleInputChange} value={security_question} />
              </div>
              <div className="form-group">
                Security Answer:
                <input type="text" name="security_answer" className="form-control security_answer" required onChange={this.handleInputChange} value={security_answer} />
              </div>

              <button type="submit" className="btn btn-danger formsave" >Sign Up</button>

              <p> Already have an account with us:</p><Link to="/login"> Login</Link>
            </form>

          </div>
        </div>
      );
    }
}
export default Register;
