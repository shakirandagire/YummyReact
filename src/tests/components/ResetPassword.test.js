import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Register from '../../components/auth/ResetPassword';

const component = shallow(<Register />);
const preventDefault = jest.fn();

describe('<ResetPassword />', () => {
  it('should render properly', () => {
    expect(component.length).toBe(1);
  });
  it('should render pagination', () => {
    expect(shallowToJson(component)).toMatchSnapshot();
  });
  it('should have reset password method', () => {
    expect(component.instance().handleResetPassword({ preventDefault }));
  });
  it('should have initial state', () => {
    expect(component.state().email).toEqual("");
    expect(component.state().new_password).toEqual("");
    expect(component.state().security_question).toEqual("");
    expect(component.state().security_answer).toEqual("");
  });
  it('should change state', () => {
    component.setState({ email: 'shakira@gmail.com', new_password: '1234567', security_question: "food", security_answer: "rice" });
    expect(component.find('[name="email"]').props().value).toEqual('shakira@gmail.com');
    expect(component.find('[name="new_password"]').props().value).toEqual('1234567');
    expect(component.find('[name="security_question"]').props().value).toEqual('food');
    expect(component.find('[name="security_answer"]').props().value).toEqual('rice');
  });
  it('renders a form', () => {
    expect(component.find('form')).toHaveLength(1);
    expect(component.find('form').simulate('submit', { preventDefault }));
    expect(preventDefault).toBeCalled();
  });
});
