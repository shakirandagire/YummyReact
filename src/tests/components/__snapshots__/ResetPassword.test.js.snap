// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResetPassword /> should render pagination 1`] = `
<div
  className="my_container backgimg"
>
  <nav
    className="navbar navbar-inverse"
  >
    <div
      className="container-fluid"
    >
      <div
        className="navbar-header"
      >
        <Link
          className="navbar-brand title"
          replace={false}
          to="/"
        >
          Yummy Recipe App
        </Link>
      </div>
    </div>
  </nav>
  <div
    className="mycontent register"
  >
    <p>
       Forgot your password:
    </p>
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        Email:
        <input
          className="form-control email"
          name="email"
          onChange={[Function]}
          required={true}
          type="email"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        New Password:
        <input
          className="form-control new_password"
          name="new_password"
          onChange={[Function]}
          required={true}
          type="password"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        Security Question:
        <input
          className="form-control security_question"
          name="security_question"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="form-group"
      >
        Security Answer:
        <input
          className="form-control security_answer"
          name="security_answer"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
      <button
        className="btn btn-danger formsave"
        type="submit"
      >
        Reset password
      </button>
    </form>
  </div>
</div>
`;
